jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +168,101 @@ public void removeComment(final String serverTemplateId, final String containerI
+        return actions;
+    }
+
+    public List<CaseActionSummary> getAllUserTaskActionSummaries(String containerId, String caseId, String userId) {
+        List<CaseActionSummary> allActionsSummaries = new ArrayList<>();
+
+        List<String> allStatus = new ArrayList<>();
+        allStatus.addAll(CaseActionStatus.AVAILABLE.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.IN_PROGRESS.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_FAIL.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_SUCCESS.getTaskStatuses());
+
+        List<NodeInstance> activeNodes = client.getActiveNodes(containerId, caseId, 0, PAGE_SIZE_UNLIMITED);
+        List<CaseActionSummary> activeNoUserTaskActionsSummaries =
+                activeNodes.stream()
+                        .filter(c -> !Arrays.asList(CaseActionNodeInstanceMapper.NODE_TYPE_HUMAN_TASK_NODE,
+                                CaseActionNodeInstanceMapper.NODE_TYPE_Dynamic_HUMAN_TASK)
+                                .contains(c.getNodeType()))
+                        .map(new CaseActionNodeInstanceMapper())
+                        .map(s -> {
+                            s.setStatus(CaseActionStatus.IN_PROGRESS.getTaskStatuses().get(0));
+                            return s;
+                        }).collect(toList());
+
+        if (!activeNoUserTaskActionsSummaries.isEmpty()) {
+            allActionsSummaries.addAll(activeNoUserTaskActionsSummaries);
+        }
+
+        List<TaskSummary> actionHumanTaskSummaries = client.findCaseTasksAssignedAsBusinessAdministrator(caseId, userId, allStatus, 0, PAGE_SIZE_UNLIMITED, "createdOn", false);
+        if (!actionHumanTaskSummaries.isEmpty()) {
+            allActionsSummaries.addAll(actionHumanTaskSummaries.stream().map(new CaseActionMapper()).collect(toList()));
+        }
+
+        return allActionsSummaries;
+    }
+
+    public List<CaseActionSummary> getAdHocActions(String serverTemplateId, String containerId, String caseId) {
+        List<CaseActionSummary> adHocActions = client.getAdHocFragments(containerId, caseId)
+                .stream()
+                .map(new CaseActionAdHocMapper())
+                .sorted(comparing(CaseActionSummary::getName)).collect(toList());
+        List<CaseActionSummary> caseAdHocActions = getCaseStagesAdHocActions(serverTemplateId, containerId, caseId);
+        if (caseAdHocActions.size() > 0) {
+            adHocActions.addAll(caseAdHocActions);
+        }
+        return adHocActions;
+    }
 
+    public List<CaseActionSummary> getCaseStagesAdHocActions(String serverTemplateId, String containerId, String caseId) {
+        CaseInstanceSummary caseInstanceSummary = getCaseInstance(serverTemplateId, containerId, caseId);
+        List<CaseActionSummary> adHocActionsActions = new ArrayList<>();
```

Comment:
stages is never null, you could use streams here to simplify, something like:
`return caseInstanceSummary.getStages().stream().flatMap(s -> s.getAcHocActions().stream()).collect(toList());`